src/Examples/UnitModels/Reactors/stoichiometric_reactor_testing.ipynb

```diff
@@ -0,0 +1,1073 @@
+    "- Objective (to define an objective function for optimization)\n",
+    "- TransformationFactory (to apply certain transformations)\n",
+    "- Arc (to connect two unit models)\n",
+    "\n",
+    "For further details on these components, please refer to the pyomo documentation: https://pyomo.readthedocs.io/en/latest/\n",
+    "\n",
+    "From idaes, we will be needing the FlowsheetBlock and the following unit models:\n",
+    "- Mixer\n",
+    "- Heater\n",
+    "- StoichiometricReactor\n",
+    "\n",
+    "We will also be needing some utility tools to put together the flowsheet and calculate the degrees of freedom, tools for model expressions and calling variable values, and built-in functions to define property packages, add unit containers to objects and define our initialization scheme.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pyomo.environ import (Constraint,\n",
+    "                           exp,\n",
+    "                           Var,\n",
+    "                           ConcreteModel,\n",
+    "                           Expression,\n",
+    "                           Objective,\n",
+    "                           TransformationFactory,\n",
+    "                           value,\n",
+    "                           units as pyunits)\n",
+    "from pyomo.network import Arc\n",
+    "\n",
+    "from idaes.core import FlowsheetBlock\n",
+    "from idaes.generic_models.properties.core.generic.generic_property import (\n",
+    "        GenericParameterBlock)\n",
+    "from idaes.generic_models.properties.core.generic.generic_reaction import (\n",
+    "        GenericReactionParameterBlock)\n",
+    "from idaes.generic_models.unit_models import (Mixer,\n",
+    "                                              Heater,\n",
+    "                                              StoichiometricReactor)\n",
+    "\n",
+    "from idaes.core.util.constants import Constants\n",
+    "from idaes.core.util import get_solver\n",
+    "from idaes.core.util.model_statistics import degrees_of_freedom\n",
+    "from idaes.core.util.initialization import propagate_state\n",
+    "\n",
+    "import idaes.logger as idaeslog"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Importing required thermophysical and reaction packages\n",
+    "\n",
+    "The final set of imports are to import the thermophysical and reaction packages. We have created a custom thermophysical package that support ideal vapor and liquid behavior for this system, and in this case we will restrict it to ideal liquid behavior only.\n",
+    "\n",
+    "The reaction package here assumes Arrhenius kinetic behavior for the StoichiometricReactor, for which $k_0$ and $E_a$ are known *a priori* (if unknown, they may be obtained using one of the parameter estimation tools within IDAES).\n",
```

Comment:
For the stoichiometric reactor, we do not use the rate constant at all so I wonder if it would be better to leave most of this out. All that gets used from the reaction package is the definition of the reactions and their stoichiometry.